src/zmq/zmqnotificationinterface.cpp

```diff
@@ -58,23 +51,22 @@ CZMQNotificationInterface* CZMQNotificationInterface::Create()
             notifier->SetType(entry.first);
             notifier->SetAddress(address);
             notifier->SetOutboundMessageHighWaterMark(static_cast<int>(gArgs.GetArg(arg + "hwm", CZMQAbstractNotifier::DEFAULT_ZMQ_SNDHWM)));
-            notifiers.push_back(notifier);
+            notifiers.emplace_back(notifier);
         }
     }
 
+    std::unique_ptr<CZMQNotificationInterface> notificationInterface;
     if (!notifiers.empty())
     {
-        notificationInterface = new CZMQNotificationInterface();
-        notificationInterface->notifiers = notifiers;
+        notificationInterface.reset(new CZMQNotificationInterface());
+        notificationInterface->notifiers = std::move(notifiers);
 
-        if (!notificationInterface->Initialize())
-        {
-            delete notificationInterface;
-            notificationInterface = nullptr;
+        if (!notificationInterface->Initialize()) {
+            notificationInterface.reset();
         }
     }
 
-    return notificationInterface;
+    return notificationInterface.release();
```

Comment:
69f92ffce9f0d11d9389610e702b99d9038f3fed

If this code is touched to make it easier to read and maintain, may I suggest a simpler approach:
```
    if (!notifiers.empty()) {
        auto notificationInterface = std::unique_ptr<CZMQNotificationInterface>(new CZMQNotificationInterface());
        notificationInterface->notifiers = std::move(notifiers);

        if (notificationInterface->Initialize()) {
            return notificationInterface.release();
        }
    }

    return nullptr;
```
?

Reply:
Indeed, looks better.  I've updated the code accordingly (except I changed the `auto notificationInterface = ...` line slightly into what I consider even clearer; but let me know if you prefer your suggestion).